containers/podman-desktop
# fix: add workaround for socket path length error

### What does this PR do?

This fixes connectivity to the Podman machine when the generated socket length is greater than 104 characters in MacOS. 

### What issues does this PR fix or reference?

Fixes problem I mentioned in a comment relating to [Issue 1128](https://github.com/containers/podman-desktop/issues/1128#issuecomment-1373913133). This logic change mimics the [behavior of the podman CLI](https://github.com/containers/podman/blob/main/pkg/machine/config.go#L303) when generating the socket path.

### How to test this PR?

It's fairly niche. You need to create a podman machine such that the following path is longer than 104 chars:

```javascript
path.resolve(path.resolve(os.homedir(), '.local/share/containers/podman/machine'), machineName, 'podman.sock');
```
Note that, the machine name is limited to 30 chars so the home directory path needs to be longer than 24 chars.

Once that is done, you should be able to connect to your podman VM via podman desktop from this branch, but it is not possible with the latest release.
